_posts/2022-01-17-changelog-163.markdown

```diff
@@ -0,0 +1,137 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #163"
+date: 2022-01-17 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#8018] Add a widget for the [external linter](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-analysis.html#call-external-linter).
+Widget indicates whether the linter is turned on. Also, the widget shows a notification when linter significantly affects the code analysis performance
+//TODO: gif
+
+* [#8132] Add `Extract trait` refactoring (`Refactor | Extract trait`)
+//TODO: gif
+
+* [#8315] Complete the second path segment if the first segment is unresolved but can be imported.
+//TODO: gif
+
+* [#8284] Provide completion for out-of-scope items when invoked at an empty place
+
+* [#6353] Detect `Const cannot refer to static` [E0013](https://doc.rust-lang.org/error-index.html#E0013) compiler error  (by [@zeroeightysix])
+
+* [#8144] Detect [E0747](https://doc.rust-lang.org/error-index.html#E0747) compiler error
+
+* [#8289] Prioritize expressions that can be implicitly coerced to the expected type when suggesting completion variants
+// I think this is rather a feature, not fix
+
+* [#8335] Do not run inspections on files disabled by `cfg` attributes
+
+* [#8323] Process available items generated by build script and expand available procedural macro calls even if the whole project host code (build scripts and procedural macro libraries) cannot be successfully compiled and executed
+
+* [#8119] Enable new name resolution inside functions bodies
+// Feature or fix?
+
+## Fixes
+
+* [#8362] Do not produce a false-positive error about missing format macro arguments without `format_args_capture` feature starting with 1.58
+
+* [#8334] Don't perform cargo refresh when another IDE project become trusted
+
+* [#8318] Expand proc macros from `dev-dependencies`. Note, procedural macro expansion is still under development, and it's disabled by default for now. To turn it on, enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental features](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
+
+* [#8297] Fix default parameters handling in `Wrong number of type arguments` inspection
+
+* [#8291] Improve sorting of auto-import variants
+// Feature?
+
+* [#8288] Fix auto-import from `alloc` crate when using [`#![no_std]`](https://docs.rust-embedded.org/book/intro/no-std.html)
+
+* [#8286] Don't show auto-import from auxiliary stdlib crates such as `unwind`
+
+* [#8279] Fix resolve of imports inside detached files
+
+* [#8269] Correctly parenthesize block-like expressions (e.g. `match`) in `Inline Value` refactoring (by [@Kobzol])
+
+* [#8268] Fix type inference if there are dependencies named the same as stdlib crates
+
+* [#8263] Infer generic parameter types in Convert closure to function intention (by [@Kobzol])
+
+* [#8230] "Introduce local variable" no longer moves the new binding into closure bodies if it is used outside of them (by [@afetisov])
```

Comment:
```suggestion
* [#8230] `Introduce local variable` refactoring no longer moves the new binding into closure bodies if it is used outside of them (by [@afetisov])
```

Reply:
Maybe also add a link to our doc: https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#extractvar-refactoring ?